### PR TITLE
refactor: Replace i18n from dcl-gatsby to react-intl

### LINKED ISSUE #1030: Replace i18n from dcl-gatsby to react-intl

### PR DESCRIPTION
**Pull request by ncomerci**
This closes #1030

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "build:server": "tsc -p .",
     "build:front": "GATSBY_VERSION_NUMBER=$VERSION_NUMBER GATSBY_HEROKU_APP_NAME=$HEROKU_APP_NAME gatsby build",
     "develop": "gatsby develop --https -H 0.0.0.0",
-    "format": "prettier --write \"**/*.{js,jsx,json,md}\"",
+    "format": "prettier --write \"**/*.{js,jsx,json,md,ts,tsx}\"",
```
Comment (andyesp) [inline package.json:15]: Are we using this script?

File: src/components/Common/Link.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import classNames from 'classnames'
+
+import { isLocalLink, navigate } from '../../utils/locations'
+
+import './Link.css'
+
+export type LinkProps = React.AnchorHTMLAttributes<HTMLAnchorElement>
+
+export default React.memo(function Link({ target, rel, href, onClick, ...props }: LinkProps) {
```
Comment (andyesp) [inline src/components/Common/Link.tsx:11]: React.memo not necessary here.

File: src/components/Common/Link.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.Link {
+  font-size: inherit;
+  line-height: inherit;
+  color: var(--primary);
```
Comment (andyesp) [inline src/components/Common/Link.css:4]: Let's use `--dcl-primary` as this is the one in our css theme.

File: src/components/Common/Link.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React, { useCallback, useMemo } from 'react'
+
+import classNames from 'classnames'
+
+import { isLocalLink, navigate } from '../../utils/locations'
+
+import './Link.css'
+
+export type LinkProps = React.AnchorHTMLAttributes<HTMLAnchorElement>
```
Comment (andyesp) [inline src/components/Common/Link.tsx:9]: I think we can remove the export.

File: src/components/Common/Link.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+export default React.memo(function Link({ target, rel, href, onClick, ...props }: LinkProps) {
+  const isLocal = useMemo(() => isLocalLink(href), [href])
+  const linkTarget = useMemo(() => (!target && !isLocal ? '_blank' : target || undefined), [isLocal, target])
```
Comment (andyesp) [inline src/components/Common/Link.tsx:13]: useMemo not necessary here.

File: src/components/Common/Link.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        navigate(href)
+      }
+    },
+    [href, isLocal, onClick]
+  )
+
+  return (
+    <a
+      {...props}
+      className={classNames('Link', (onClick || href) && 'Link--pointer', props.className)}
```
Comment (andyesp) [inline src/components/Common/Link.tsx:31]: You can add className to the destructuring too.

File: src/components/Common/Link.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import React, { useCallback, useMemo } from 'react'
+
+import classNames from 'classnames'
+
+import { isLocalLink, navigate } from '../../utils/locations'
```
Comment (andyesp) [inline src/components/Common/Link.tsx:5]: If we are not reusing this function I'd leave it here.

File: src/components/Common/Link.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+import './Link.css'
+
+export type LinkProps = React.AnchorHTMLAttributes<HTMLAnchorElement>
+
+export default React.memo(function Link({ target, rel, href, onClick, ...props }: LinkProps) {
+  const isLocal = useMemo(() => isLocalLink(href), [href])
+  const linkTarget = useMemo(() => (!target && !isLocal ? '_blank' : target || undefined), [isLocal, target])
+  const linkRel = useMemo(() => (!isLocal ? classNames(rel, 'noopener', 'noreferrer') : rel), [isLocal, rel])
```
Comment (andyesp) [inline src/components/Common/Link.tsx:14]: Are we passing a different type of `rel`? If we are not then I'd change this to `const linkRel = !isLocal ? "noopener noreferrer" : undefined`

File: package.json [hunk 9]
```diff
@@ -33,7 +33,8 @@
       "<rootDir>/src/"
     ],
     "moduleNameMapper": {
-      "\\.(css|less|sass|scss|gif|ttf|eot|svg)$": "<rootDir>/src/__mocks__/files.ts"
+      "\\.(css|less|sass|scss|gif|ttf|eot|svg)$": "<rootDir>/src/__mocks__/files.ts",
+      "^gatsby-page-utils/(.*)$": "gatsby-page-utils/dist/$1"
```
Comment (andyesp) [inline package.json:37]: What is this for?

File: src/components/Status/StatusPill.tsx [hunk 10]
```diff
@@ -37,7 +37,7 @@ const StatusPill = ({ className, status, size, isLink }: Props) => {
   const icon = showIcon ? <Check color={iconColor} /> : null
 
   const Wrapper = isLink ? Link : 'div'
-  const href = isLink ? locations.proposals({ status: status }) : undefined
+  const href = isLink ? locations.proposals({ status: status }) : ''
```
Comment (andyesp) [inline src/components/Status/StatusPill.tsx:40]: If this is an empty string the 'div' will show with an empty `href`.

File: package.json (hunk 1 above)
Comment (ncomerci) [reply package.json:15]: I've used it to replace imports faster

File: src/components/Common/Link.tsx (hunk 8 above)
Comment (ncomerci) [reply src/components/Common/Link.tsx:14]: Here

image

File: gatsby-node.js [hunk 11]
```diff
@@ -22,4 +22,17 @@ exports.onCreateWebpackConfig = ({ actions, plugins }) => {
...
+  // This is only an issue with regular CSS being imported.
+  if (stage === 'build-javascript' || stage === 'develop') {
+    const config = getConfig()
+    // Get the mini-css-extract-plugin
+    const miniCssExtractPlugin = config.plugins.find((plugin) => plugin.constructor.name === 'MiniCssExtractPlugin')
+    // Set the option here to true.
+    if (miniCssExtractPlugin) {
+      miniCssExtractPlugin.options.ignoreOrder = true
+    }
+    // Update the config.
```
Comment (1emu) [inline gatsby-node.js:35]: I don't think we need this gepetty comments here 😄 , maybe just the first two

**Review by 1emu — APPROVED** [review]

File: gatsby-node.js (hunk 11 above)
Comment (ncomerci) [reply gatsby-node.js:35]: they are Frami comments haha
https://github.com/decentraland/decentraland-gatsby/blob/master/src/plugins/intl/gatsby-node.ts#L96

**Review by 1emu — APPROVED** [review]

**Review by andyesp — APPROVED** [review]

**Pull request merged**